sdk/eventhub/azure-eventhub/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,228 @@
+
+For more information about the `Azure.Identity` library, check out our [Authentication and the Azure SDK][AuthenticationAndTheAzureSDK] blog post.
+
+## Enable and configure logging
+
+The Azure SDK for Python offers a consistent logging story to help troubleshoot application errors and expedite their resolution.  The logs produced will capture the flow of an application before reaching the terminal state to help locate the root issue.
+
+This library uses the standard [Logging] library for logging
+
+- Enable `azure.eventhub` logger to collect traces from the library.
+- Enable `uamqp` logger to collect traces from the underlying uAMQP library.
+
+### Enable AMQP transport logging
+
+If enabling client logging is not enough to diagnose your issues. You can enable AMQP frame level trace by setting `logging_enable=True` when creating the client.
+
+### Reduce logging
+
+There may be cases where you consider the `uamqp` logging to be too verbose. To suppress unnecessary logging, add the following snippet to the top of your code:
+
+```python
+import logging
+
+# The logging levels below may need to be adjusted based on the logging that you want to suppress.
+uamqp_logger = logging.getLogger('uamqp')
+uamqp_logger.setLevel(logging.ERROR)
+
+# or even further fine-grained control, suppressing the warnings in uamqp.connection module
+uamqp_connection_logger = logging.getLogger('uamqp.connection')
+uamqp_connection_logger.setLevel(logging.ERROR)
+```
+
+## Troubleshoot EventProducerAsyncClient/EventProducerClient issues
+
+### Cannot set multiple partition keys for events in EventDataBatch
+
+When publishing messages, the Event Hubs service supports a single partition key for each EventDataBatch.  Customers can consider using the producer client in `buffered mode` if they want that capability.  Otherwise, they'll have to manage their batches.
+
+### Setting partition key on EventData is not set in Kafka consumer
+
+The partition key of the EventHubs event is available in the Kafka record headers, the protocol specific key being "x-opt-partition-key" in the header.
+
+By design, Event Hubs does not promote the Kafka message key to be the Event Hubs partition key nor the reverse because with the same value, the Kafka client and the Event Hub client likely send the message to two different partitions.  It might cause some confusion if we set the value in the cross-protocol communication case.  Exposing the properties with a protocol specific key to the other protocol client should be good enough.
+
+## Troubleshoot EventProcessorClient issues
+
+### 412 precondition failures when using an event processor
+
+412 precondition errors occur when the client tries to take or renew ownership of a partition, but the local version of the ownership record is outdated.  This occurs when another processor instance steals partition ownership.  See [Partition ownership changes a lot](#partition-ownership-changes-a-lot) for more information.
```

Comment:
It's an HTTP 412 response from the Storage service.   For most languages, that gets logged by Azure.Core and surfaced in AppInsights, it's not directly observable from the Event Hubs clients in most languages since it's part of normal operation. 

I'm not sure that Python has the same logging behavior or AI integration, though....

Reply:
The .NET troubleshooting guide went with "Logs reflect intermittent HTTP 412 and HTTP 409 responses from storage" for this to try to clarify where you'd see the errors.    We tend to get a lot of tickets for this because AppInsights displays this as an error.

Reply:
I like the .NET wording for the HTTP response and the intent from the Exception / log message